Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1633,6 +1634,358 @@
     UNLOCK_CAIRO();
 }
 
+void CGContextCairo::CGContextReplacePathWithStrokedPath(CGContextRef context) {
+    ObtainLock();
+
+    LOCK_CAIRO();
+
+    // Create a flat path from the current path. This will contain no curves
+    cairo_path_t* flatPath = cairo_copy_path_flat(_drawContext);
+    // Start working with a new path
+    cairo_new_path(_drawContext);
+
+    CGFloat halfStrokeWidth = (curState->lineWidth) / 2.0f;
+
+    // Calculates points on stroked path from original points
+    // Reverse flag should be true if we're traversing the path backwards
+    auto strokedPoint = [](CGPoint p, CGFloat angle, CGFloat distance, bool reverse) -> CGPoint {
+        if (reverse) {
+            angle += M_PI;
+        }
+        CGFloat x = p.x - (sin(angle) * distance);
+        CGFloat y = p.y + (cos(angle) * distance);
```

Comment:
Right, this should be equivalent to:

`CGFloat x = p.x + (cos(angle - M_PI_2) * distance);`
`CGFloat y = p.y + (sin(angle - M_PI_2) * distance);`

Would you prefer this implementation so it appears more straightforward instead of comments?